src/kani-compiler/rustc_codegen_kani/src/codegen/statement.rs

```diff
@@ -78,18 +79,24 @@ impl<'tcx> GotocCtx<'tcx> {
                     if *expected { r } else { Expr::not(r) }
                 };
 
-                Stmt::block(
-                    vec![
-                        cond.cast_to(Type::bool()).if_then_else(
-                            Stmt::goto(self.current_fn().find_label(target), loc.clone()),
-                            None,
-                            loc.clone(),
-                        ),
-                        Stmt::assert_false(&format!("{:?}", msg), loc.clone()),
+                let mut msg_str = format!("{:?}", msg);
+                let mut stmts: Vec<Stmt> = Vec::new();
```

Comment:
Do these need to be mutable?

Reply:
Making them mutable was the only way I could think of to avoid code duplication. I'm open for suggestions.

Reply:
I changed them to immutable by using the `let x = if {...} else {...};` pattern in which I used the `Stmt::Skip` type. Let me know if this looks better.